bson-scala/src/main/scala/org/mongodb/scala/bson/collection/BaseDocument.scala

```diff
@@ -242,7 +242,7 @@ private[bson] trait BaseDocument[T] extends Traversable[(String, BsonValue)] wit
   /**
    * Returns a copy of the underlying BsonDocument
    */
-  def toBsonDocument: BsonDocument = copyBsonDocument()
+  override def toBsonDocument: BsonDocument = copyBsonDocument()
```

Comment:
I'm not sure about this override.  

`toBsonDocument[TDocument](documentClass: Class[TDocument], codecRegistry: CodecRegistry): BsonDocument`  returns `this`, essentially

But `BaseDocument#toBsonDocument `returns a copy of itself

But the intention in `Bson` is to make those two methods essentially identical in behavior

Reply:
I think thats fine, it should return `underlying`.  Note that `ImmutableDocumentCodec#generateIdIfAbsentFromDocument` will have to be updated to clone the document so it doesn't mutate the original:

```
  override def generateIdIfAbsentFromDocument(document: Document): Document = {
    if (!underlying.documentHasId(document.underlying)) {
      Document(underlying.generateIdIfAbsentFromDocument(document.toBsonDocument.clone()))
    } else {
      document
    }
  }
```

Reply:
The doc comment says that it returns a copy.  Could this change break clients that are assuming that it returns a copy?

Reply:
I doubt users use the method, its really there for the codecs. Alternatively, you'll have to leave as is.

Reply:
Thanks.  I looked for other usages and found:

```
trait LowPrio {

  /**
   * Transforms `immutable.Document` to `BsonDocument`
   */
  implicit object TransformImmutableDocument extends BsonTransformer[IDocument] {
    def apply(value: IDocument): BsonDocument = value.toBsonDocument
  }
   ...
```

Does that have to change as well?

Reply:
No, thats fine

Reply:
Done